internal/pkg/deploy/cloudformation/stack/backend_svc.go

```diff
@@ -201,6 +207,27 @@ func (s *BackendService) Template() (string, error) {
 	return string(overriddenTpl), nil
 }
 
+func (s *BackendService) containerPortMappings() ([]*template.PortMapping, []*template.PortMapping, error) {
```

Comment:
how about making the output to be a map? so the primary container port mapping is `output[workloadName]`. And also we can make `convertSidecar` much more efficient.